server/planning/feed_parsers/superdesk_planning_xml.py

```diff
@@ -144,11 +151,17 @@ def parse_content_meta(self, tree, item):
         slugline_elt = content_meta.find(self.qname("headline"))
         if slugline_elt is not None and slugline_elt.text:
             item["slugline"] = slugline_elt.text
-            item["name"] = slugline_elt.text
+            item["name"] = (
+                plain_text_to_html(slugline_elt.text) if is_field_editor_3("name", planning_type) else slugline_elt.text
+            )
 
         description_elt = content_meta.find(self.qname("description"))
         if description_elt is not None and description_elt.text:
-            item["description_text"] = description_elt.text
+            item["description_text"] = (
```

Comment:
Rather than manually updating logic for each field, you can add a new method that can be called at the end of the `parse` method.

Such as:
```python
# file: planning/content_profiles/utils.py
def get_editor3_fields(resource: str) -> Set[str]:
    content_type = get_planning_schema(resource)
    resource_schema = content_type.get("schema") or {}

    return set(
        field_name
        for field_name, field_schema in resource_schema.items()
        if (
            is_field_enabled(field_name, content_type)
            and (field_schema or {}).get("field_type") == "editor_3"
        )
    )
```

```python
# file: planning/feed_parsers/utils.py
from planning.content_profiles.utils import get_editor3_fields

def upgrade_rich_text_fields(item, resource: str):
    item.update({
        field: plain_text_to_html(item[field])
        for field in get_editor3_fields(resource)
        if item.get(field)
    })
```

```python
# file: planning/feed_parsers/superdesk_planning_xml.py
from .utils import upgrade_rich_text_fields

class PlanningMLParser(NewsMLTwoFeedParser):
    def parse(self, tree: Element, provider=None):
        item = {...}
        upgrade_rich_text_fields(item, "planning")
        for coverage in item.get("coverages") or []:
            upgrade_rich_text_fields(coverage.get("planning") or {}, "coverages")
```

```python
# file: planning/feed_parseres/events_ml.py
from .utils import upgrade_rich_text_fields

class EventsMLParser(NewsMLTwoFeedParser):
    def parse(self, tree: Element, provider=None):
        item = {...}
        upgrade_rich_text_fields(item, "event")
```